### PR TITLE
rules_libusb@0.1.0-rc1

### PR DESCRIPTION
**Pull request by armandomontanez**
First commit to add Pigweed's rules_libusb module to the BCR. We work from Gerrit, which does not have stable tarball hashes, so releases are provided via GCS.

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (rules_libusb) have been updated in this PR. Please review the changes.

**Comment by armandomontanez** [comment]
Current failure is:
```
BcrValidationResult.FAILED: The source URL of rules_libusb@0.1.0-rc1 (https://storage.googleapis.com/storage/v1/b/pigweed-bazel-tarballs/o/rules_libusb-8745211641617303569-f5c0af2f037b6e8405b4158b5e3041675da8e522.tar.gz?alt=media) doesn't match any of the module's source repositories ['https://pigweed.googlesource.com/pigweed/rules_libusb/'].
```

Is there a way we can allowlist this particular instance? The only alternative is expressing this dependency as a git_repository, which isn't ideal. We have an automated builder that creates these, the first number is the buildbucket ID and the second is the commit hash of interest.

Separately, it appears the tools are choking on the URL and ignore the archive_type hint:
```
Traceback (most recent call last):
--
  | File "./tools/bcr_validation.py", line 407, in <module>
  | sys.exit(main())
  | File "./tools/bcr_validation.py", line 396, in main
  | validator.validate_module(name, version, args.skip_validation)
  | File "./tools/bcr_validation.py", line 301, in validate_module
  | self.verify_module_dot_bazel(module_name, version)
  | File "./tools/bcr_validation.py", line 217, in verify_module_dot_bazel
  | shutil.unpack_archive(str(archive_file), output_dir)
  | File "/usr/lib/python3.8/shutil.py", line 1237, in unpack_archive
  | raise ReadError("Unknown archive format '{0}'".format(filename))
  | shutil.ReadError: Unknown archive format '/tmp/tmppdhe8g_g/rules_libusb-8745211641617303569-f5c0af2f037b6e8405b4158b5e3041675da8e522.tar.gz?alt=media'
```

File: modules/rules_libusb/metadata.json
Comment (Wyverald) [inline modules/rules_libusb/metadata.json:22]: you can just remove this line if your source archives aren't going to come from here.

**Comment by armandomontanez** [comment]
Nice, now it's just stuck on the unknown archive format:
```
shutil.ReadError: Unknown archive format '/tmp/tmptdjdzfta/rules_libusb-8745211641617303569-f5c0af2f037b6e8405b4158b5e3041675da8e522.tar.gz?alt=media'
```
Unfortunately, if I drop the `?alt=media` then it fetches the wrong thing. The https://storage.cloud.google.com version of the URL also fetches the wrong thing.

**Comment by armandomontanez** [comment]
Sent out https://github.com/bazelbuild/bazel-central-registry/pull/2261 to work around the current failure.

**Comment by armandomontanez** [comment]
Should be ready for another look now, everything that has run is passing.

**Comment by Wyverald** [comment]
Sorry, looks like there are more places in the BCR presubmit script that need patching :S

File: modules/rules_libusb/0.1.0-rc1/presubmit.yml
Comment (Wyverald) [inline modules/rules_libusb/0.1.0-rc1/presubmit.yml:12]: looks like this should be `@libusb//:libusb_dynamic`. alternatively, you could add an `alias` target in `rules_libusb` itself.

File: modules/rules_libusb/0.1.0-rc1/presubmit.yml
Comment (Wyverald) [inline modules/rules_libusb/0.1.0-rc1/presubmit.yml:5]: 6.x don't support not having a WORKSPACE file.

**Comment by armandomontanez** [comment]
Filed https://pwbug.dev/347353308 to track failing workflows. I'll cut a second release candidate after I've had a chance to verify other environments after RC1 merges.

**Review by Wyverald — APPROVED** [review]

**Pull request merged**
